backend/src/main/java/com/eyetrade/backend/service/CurrencyConverterService.java

```diff
@@ -0,0 +1,63 @@
+package com.eyetrade.backend.service;
+
+import com.eyetrade.backend.constants.CurrencyType;
+import com.eyetrade.backend.model.dto.CurrencyConverterDto;
+import com.eyetrade.backend.model.entity.CurrencyRecord;
+import com.eyetrade.backend.model.resource.CurrencyConverterResource;
+import com.eyetrade.backend.repository.CurrencyRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.Date;
+
+import static com.eyetrade.backend.constants.CurrencyConstants.CURRENCY_EXPIRE_TIME;
+
+/**
+ * Created by Emir Gökdemir
+ * on 17 Eki 2019
+ */
+@Service
+public class CurrencyConverterService {
+
+    @Autowired
+    private CurrencyRepository currencyRepository;
+
+    @Autowired
+    private CurrencyRecordService currencyRecordService;
+
+    public CurrencyConverterResource convertCurrency (CurrencyConverterDto converterDto){
+        CurrencyRecord record= currencyRepository.findTopByOrderByTimestamp();
+        if(record==null || checkExpired(record.getTimestamp())){
+            try {
+                record=currencyRecordService.getCurrencyRecord();
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Printing the stack trace is dangerous in terms of security. Instead of that, we can raise an exception with a message.

Reply:
Also, we should add error handling in the future to return proper error message to the client application